core/org.wso2.carbon.ui/src/main/resources/web/admin/login.jsp

```diff
@@ -109,9 +110,11 @@ String bannerContent = adminConfig.getBannerContent();
     <%
         String loginStatus = CharacterEncoder.getSafeText(request.getParameter("loginStatus"));
         String errorCode = CharacterEncoder.getSafeText(request.getParameter("errorCode"));
+        String BUNDLE = "org.wso2.carbon.i18n.Resources";
 
         if (loginStatus != null && "false".equalsIgnoreCase(loginStatus)) {
-            if (errorCode == null) {
+            if (errorCode == null || (!(StringUtils.isBlank(errorCode)) &&
+                    errorCode.equalsIgnoreCase(CarbonUIUtil.geti18nString(errorCode, BUNDLE, request.getLocale())))) {
```

Comment:
`errorCode == null` check is redundant
```suggestion
            if (StringUtils.isNotBlank(errorCode) && 
                                       errorCode.equalsIgnoreCase(CarbonUIUtil.geti18nString(errorCode, BUNDLE, request.getLocale()))) {
```

Reply:
The null check is also required since that was the initial behaviour and it has to be preserved.

Reply:
`StringUtils.isNotBlank(errorCode)` would contain the null check as well. Hence we can simplify the code here

Reply:
Hence the condition requires execution even if error code is null, the operation is updated to OR as discussed.